core/templates/services/user.service.ts

```diff
@@ -56,21 +62,37 @@ export class UserService {
     return this.userInfo;
   }
 
-  async getProfileImageDataUrlAsync(): Promise<string> {
-    let defaultUrl = (
-      this.urlInterpolationService.getStaticImageUrl(
-        AppConstants.DEFAULT_PROFILE_IMAGE_PATH));
-    return this.getUserInfoAsync().then(
-      async(userInfo) => {
-        if (userInfo.isLoggedIn()) {
-          return this.userBackendApiService.getProfileImageDataUrlAsync(
-            defaultUrl);
-        } else {
-          return new Promise((resolve, reject) => {
-            resolve(defaultUrl);
+  private async _getProfileImagePromise(imageDataBlob: Blob): Promise<string> {
+    return new Promise(resolve => {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        resolve(reader.result as string);
+      };
+      reader.readAsDataURL(imageDataBlob);
+    });
+  }
+
+  async getProfileImageDataUrlAsync(username: string = ''): Promise<string> {
+    let defaultUrl = this.urlInterpolationService.getStaticImageUrl(
+        AppConstants.DEFAULT_PROFILE_IMAGE_PATH);
+     return this.getUserInfoAsync().then(userInfo => {
+      if (username === '') {
+        username = userInfo.getUsername();
+      }
```

Comment:
I think we would require this. For example, in blog-dashboard-page.component.ts we would require a username, now the way to get that is via `blogDashboardData` and we are doing this part in the `initMainTab()` but the first problem is that we would require to check both the draft and published blogs for the authorUsername and the second problem that I'm seeing is that whenever I'm trying to print the `dashboardData` I'm getting the `authorUsername` as undefined. You can check the below pic for reference:
![Screenshot from 2023-01-13 23-34-40](https://user-images.githubusercontent.com/44300735/212389456-a5c7a902-8c25-4bdf-8543-8aee24f6253f.png)

I think for this purpose it would make sense to get the username at this place only.
Thanks